codec-http/src/test/java/io/netty/handler/codec/http/HttpHeadersTest.java

```diff
@@ -76,8 +76,8 @@ public void testAddSelf() {
         headers.add(headers);
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testSetSelf() {
+    @Test
```

Comment:
I agree and I'm fine with your change as long as you also back port it to 4.0 ;)